framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -311,8 +422,168 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		return false;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _createCCU() {}
+	private async _calculateInboxUpdate(sideChainID: Buffer): Promise<InboxUpdate> {
+		const serialzedCCMs = this._chainConnectorState.crossChainMessages.map(ccm =>
+			codec.encode(ccmSchema, ccm),
+		);
+		const ccmHashes = serialzedCCMs.map(ccm => utils.hash(ccm));
+
+		const channel = await this._mainchainAPIClient.invoke<ChannelData>(
+			'interoperability_getChannel',
+			{ sideChainID },
+		);
+
+		let newInboxRoot;
+		let newInboxAppendPath = channel.inbox.appendPath;
+		let newInboxSize = channel.inbox.size;
+
+		for (const ccm of ccmHashes) {
+			const { appendPath, size, root } = regularMerkleTree.calculateMerkleRoot({
+				value: ccm,
+				appendPath: newInboxAppendPath,
+				size: newInboxSize,
+			});
+			newInboxAppendPath = appendPath;
+			newInboxSize = size;
+			newInboxRoot = root;
+		}
+
+		const db = new InMemoryDatabase();
+
+		const smt = new SparseMerkleTree({
+			db,
+			rootHash: channel.inbox.root,
+		});
+
+		const outboxKey = this._rawStateStoreKey(sideChainID);
+
+		await smt.update(outboxKey, newInboxRoot as Buffer);
+
+		const proofSMT = await smt.generateSingleProof(outboxKey);
+
+		const outboxRootWitness: OutboxRootWitness = {
+			bitmap: Buffer.from(proofSMT.binaryBitmap),
+			siblingHashes: proofSMT.siblingHashes,
+		};
+
+		const inboxUpdate: InboxUpdate = {
+			crossChainMessages: serialzedCCMs,
+			messageWitness: {
+				partnerChainOutboxSize: BigInt(0),
+				siblingHashes: [],
```

Comment:
this is only the case if you are posintg all pending CCMs in the CCU, which is not always possible (there is a max ccu size)

Reply:
coz of this reason https://github.com/LiskHQ/lisk-sdk/pull/7438#discussion_r957484065 we don't have partial CCU updates to keep it simple right now thats why this `messageWitness` will be empty

Reply:
But before we release we definitely need to have this feature, otherwise a chain could be stuck by just sending enough CCMs

Reply:
Issue created to tackle it for `v6.0.0` https://github.com/LiskHQ/lisk-sdk/issues/7458